### PR TITLE
testsuite: update SNI test resource checks

### PR DESCRIPTION
**Pull request by jpeach**
Adopt the standard resources policies to check whether HTTPProxy resources
were created and were marked valid.

This updates #2743.

Signed-off-by: James Peach <jpeach@vmware.com>

### CODE REVIEW

File: _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml [hunk 1]
```diff
@@ -231,42 +214,25 @@ spec:
...
 
 fatal_proxy_is_not_valid[msg] {
-  proxy := data.resources.httpproxies[Name]
-  fqdn := proxy.spec.virtualhost.fqdn
-  status := proxy.status
+  status := resources.status("httpproxies", "echo-two")
+  validity := object.get(status, "currentStatus", "not-present")
 
-  not proxy_is_valid
+  not validity == "valid"
```
Comment (stevesloka) [inline _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml:232]: So this expects the validity to "not" be `valid`?

**Review by skriss — APPROVED** [review]
Looks OK to me

File: _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml (hunk 1 above)
Comment (jpeach) [reply _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml:232]: yep, it's an error in that case.

Comment (stevesloka) [reply _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml:232]: Maybe change `validity` to `resultStatus` or something such that it reads: `not resultStatus == "valid"` or check on the reverse such that it's `invalid`.

**Review by stevesloka — CHANGES_REQUESTED** [review]

File: _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml (hunk 1 above)
Comment (jpeach) [reply _integration/testsuite/httpproxy/004-https-sni-enforcement.yaml]: I renamed it to `resultStatus`.

**Comment by jpeach** [comment]
Failure is:
```
not ok 25 - running Rego check lines ./_integration/testsuite/httpproxy/004-https-sni-enforcement.yaml:116-129
  ---
  - Message: |
      raised predicate "fatal_proxy_is_not_valid"
      HTTP proxy "echo-one" status is "not-present"
      {}
    Severity: Fatal
    Timestamp: "0001-01-01T00:00:00Z"
```

This would happen if Contour has never updated the status of the proxy.

**Review by stevesloka — APPROVED** [review]
/lgtm

**Pull request merged**